### PR TITLE
Adjust course card infos

### PR DESCRIPTION
**Pull request by LomyW**
https://github.com/corona-school/project-user/issues/595#issuecomment-1484141424

### CODE REVIEW

File: src/Utility.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ export const getTrafficStatusText = (subcourse: { course: { courseState?: Course
     if (!subcourse.published && subcourse.course.courseState === Course_Coursestate_Enum.Created) return i18next.t('single.global.courseState.draft');
     if (!subcourse.published && subcourse.course.courseState === Course_Coursestate_Enum.Submitted) return i18next.t('single.global.courseState.submitted');
     if (!subcourse.published && subcourse.course.courseState === Course_Coursestate_Enum.Allowed) return i18next.t('single.global.courseState.draft');
+    if (subcourse.course.courseState === Course_Coursestate_Enum.Denied) return i18next.t('single.global.courseState.denied');
```
Comment (Jonasdoubleyou) [inline src/Utility.ts:136]: All courseStates could be moved into if(!subcourse.published)

File: src/pages/subcourse/AllSubcourses.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const AllSubcourses: React.FC<GroupProps> = ({ languageCourses, courses, focusCo
                 showStatus={(subcourse as any)?.isInstructor ?? false}
                 trafficLightStatus={getTrafficStatus(subcourse.participantsCount || 0, subcourse.maxParticipants || 0)}
                 onPressToCourse={readonly ? undefined : () => navigate(`/single-course/${subcourse.id}`)}
-                showSchoolclass
+                showSchoolclass={(subcourse as any)?.isInstructor || (subcourse as any)?.isParticipant ? true : false}
```
Comment (Jonasdoubleyou) [inline src/pages/subcourse/AllSubcourses.tsx:47]: ? true : false could be ?? false

**Review by Jonasdoubleyou — APPROVED** [review]

File: src/pages/subcourse/AllSubcourses.tsx (hunk 2 above)
Comment (LomyW) [reply src/pages/subcourse/AllSubcourses.tsx:47]: `??` doesn't work with `||` , or?

Comment (Jonasdoubleyou) [reply src/pages/subcourse/AllSubcourses.tsx:47]: Yeah but isInstructor and isParticipant are already booleans, so we must only handle the null/undefined case ...

File: src/Utility.ts [hunk 3]
```diff
@@ -130,9 +130,12 @@ export const getTrafficStatus: (participants: number, maxParticipants: number) =
...
-    if (!subcourse.published && subcourse.course.courseState === Course_Coursestate_Enum.Submitted) return i18next.t('single.global.courseState.submitted');
-    if (!subcourse.published && subcourse.course.courseState === Course_Coursestate_Enum.Allowed) return i18next.t('single.global.courseState.draft');
+    if (!subcourse.published) {
+        if (subcourse.course.courseState === Course_Coursestate_Enum.Created) return i18next.t('single.global.courseState.draft');
+        if (subcourse.course.courseState === Course_Coursestate_Enum.Submitted) return i18next.t('single.global.courseState.submitted');
+        if (subcourse.course.courseState === Course_Coursestate_Enum.Allowed) return i18next.t('single.global.courseState.draft');
+        if (subcourse.course.courseState === Course_Coursestate_Enum.Denied) return i18next.t('single.global.courseState.denied');
+    }
     if (subcourse.published) return i18next.t('single.global.courseState.publish');
     if (subcourse.cancelled) return i18next.t('single.global.courseState.cancelled');
```
Comment (Jonasdoubleyou) [inline src/Utility.ts:140]: While looking at this, I'm not sure line 137 is actually reachable ... A cancelled course should also still be published, so it should hit 136 ... Could you check this in a followup task?

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**